src/lib/realmdb/__tests__/UserProfileStorage.js

```diff
@@ -0,0 +1,455 @@
+  username: 'juliankobrynski',
+  mobile: '+48507471353',
+  walletAddress: '0x740E22161DEEAa60b8b0b5cDAAA091534Ff21649',
+
+  // smallAvatar: '',
+}
+
+const profileKeyValue = {
+  // avatar: '',
+  email: 'julian@gooddollar.org',
+  fullName: 'Julian Kobryński',
+  mnemonic: 'duty disorder rocket velvet later fabric scheme paddle remove phone target medal',
+  username: 'juliankobrynski',
+  mobile: '+48507471353',
+  walletAddress: '0x740E22161DEEAa60b8b0b5cDAAA091534Ff21649',
+
+  // smallAvatar: '',
+}
+
+describe('UserProfileStorage', () => {
+  let userProfileStorage
+
+  beforeAll(async () => {
+    await AsyncStorage.setItem(
+      'GD_jwt',
+      'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJsb2dnZWRJbkFzIjoiMHg1YjliNDlmZjM1ZmE4OWZkMWZiOWNmNGJmNTNkNmI1MDA5ZmVjNjgxIiwiZ2RBZGRyZXNzIjoiMHg3NDBlMjIxNjFkZWVhYTYwYjhiMGI1Y2RhYWEwOTE1MzRmZjIxNjQ5IiwicHJvZmlsZVB1YmxpY2tleSI6IjlZdFNlSXdELVN3Z080UVIxaHBobGt4dFhleUdESjFIX01PQ3pncWcwWEkuTDN3RTJZUkpOT3c0cUo1UFVST0lRNTk3OVR3RFlCcmFmZGUwTlFkXzFSUSIsImV4cCI6MTYyOTA5Njg3MSwiYXVkIjoicmVhbG1kYl93YWxsZXRfZGV2ZWxvcG1lbnQiLCJzdWIiOiIweDViOWI0OWZmMzVmYTg5ZmQxZmI5Y2Y0YmY1M2Q2YjUwMDlmZWM2ODEiLCJpYXQiOjE2Mjg0OTIwNzF9.foR0GWU-63uJ4JIx1wBKgHs8FYVUNJvkirTWj-VCWGU',
+    )
+    const db = getDB()
+    await goodWallet.ready
+    userProfileStorage = new UserProfileStorage(goodWallet, db)
+    const seed = goodWallet.wallet.eth.accounts.wallet[goodWallet.getAccountForType('gundb')].privateKey.slice(2)
+    await db.init(seed, goodWallet.getAccountForType('gundb')) //only once user is registered he has access to realmdb via signed jwt
+  })
+
+  beforeEach(() => {
+    jest.restoreAllMocks()
+  })
+
+  it('should not save profile without email to the db', async () => {
+    const { email, ...fields } = profile
+
+    try {
+      await userProfileStorage.setProfile(fields)
+    } catch (exception) {
+      expect(exception).toEqual(expect.objectContaining({ email: 'Email is required' }))
+    }
+  })
+
+  it('should not save profile without mobile to the db', async () => {
+    const { mobile, ...fields } = profile
+
+    try {
+      await userProfileStorage.setProfile(fields)
+    } catch (exception) {
+      expect(exception).toEqual(expect.objectContaining({ mobile: 'Mobile is required' }))
+    }
+  })
+
+  it('should not save profile without username to the db', async () => {
+    const { username, ...fields } = profile
+
+    try {
+      await userProfileStorage.setProfile(fields)
+    } catch (exception) {
+      expect(exception).toEqual(expect.objectContaining({ username: 'Username is required' }))
+    }
+  })
+
+  it('should save profile to the db', async () => {
+    await userProfileStorage.setProfile(profile)
+
+    const encryptedProfile = await userProfileStorage.profiledb.getProfile()
+    const { user_id, _id, ...fields } = encryptedProfile
+
+    // Check if values were encrypted
+    Object.keys(fields).forEach(key => {
+      expect(profile[key]).not.toEqual(encryptedProfile[key].value)
```

Comment:
this is not a good test
what if  value is simply empty?
also it doesnt test what the test says, you need to save  profile then read it then make sure it has all the values originally saved, that  will test that it was actually saved to the database